### PR TITLE
DEPR-127: Convert from EdxRestApiClient to OAuthAPIClient

### PR DESCRIPTION
**Pull request by mikix**

### CODE REVIEW

File: credentials/apps/core/models.py [hunk 1]
```diff
@@ -252,17 +210,6 @@ class User(AbstractUser):
 
     full_name = models.CharField(_("Full Name"), max_length=255, blank=True, null=True)
 
-    @property
-    def access_token(self):
```
Comment (mikix) [inline credentials/apps/core/models.py:256]: This didn't seem used anywhere? So I just dropped it. Not necessary as part of my conversion, I can add it back if it is necessary. Just some unrelated cleanup I noticed.

File: credentials/apps/core/tests/test_models.py [hunk 2]
```diff
@@ -20,20 +19,6 @@ def setUp(self):
...
-    def test_access_token_without_social_auth(self):
-        """ Verify the property returns None if the user is not associated with a UserSocialAuth. """
-        self.assertIsNone(self.user.access_token)
-
-    def test_access_token(self):
-        """ Verify the property returns the value of the access_token stored with the UserSocialAuth. """
-        social_auth = UserSocialAuth.objects.create(user=self.user, provider="test", uid=self.user.username)
-        self.assertIsNone(self.user.access_token)
-
-        access_token = "My voice is my passport. Verify me."
```
Comment (mikix) [inline credentials/apps/core/tests/test_models.py:32]: I do love this reference.

File: credentials/apps/core/models.py (hunk 1 above)
Comment (staubina) [reply credentials/apps/core/models.py:256]: I am not 100% sure, but I think this method needs to exist for oauth. @crice100 Any idea if this is needed?

Comment (crice100) [reply credentials/apps/core/models.py:256]: I don't know, sorry

**Review by tuchfarber — APPROVED** [review]

**Pull request merged**